File: app/src/main/java/com/example/aac/java/model/TestDataViewModel.java
package com.example.aac.java.model;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.content.Context;
import android.os.SystemClock;
import android.util.Log;


import com.alibaba.fastjson.TypeReference;
import com.example.aac.java.data.activity.bean.UserBean;
import com.example.aac.utils.JsonCallback;
import com.lzy.okgo.OkGo;
import com.lzy.okgo.callback.StringCallback;
import com.lzy.okgo.model.Response;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by yangc on 2017/8/14.
 * E-Mail:yangchaojiang@outlook.com
 * Deprecated:
 */

public class TestDataViewModel extends AndroidViewModel {
    private MutableLiveData<String> data = new MutableLiveData<>();
    private MutableLiveData<List<String>> listData = new MutableLiveData<>();
    private MutableLiveData<UserBean> liveData = new MutableLiveData<>();
    public TestDataViewModel(Application application) {
        super(application);
        Log.d("TestDataViewModel", "TestDataViewModel");

    }

    public LiveData<String> getData() {
        Log.d("TestDataViewModel", data.getValue() + "");
        Thread s = new Thread(new Runnable() {
            @Override
            public void run() {
                SystemClock.sleep(1000);
                data.postValue(null);
                //data.postValue("获取数据");
            }
        });
        s.start();
        return data;
    }


    @Override
    protected void onCleared() {
        super.onCleared();
        Log.d("TestDataViewModel", "onCleared");
    }

    public LiveData<List<String>> getListData(final int page) {
        Thread s = new Thread(new Runnable() {
            @Override
            public void run() {
                SystemClock.sleep(2000);
                int s = (page - 1) * 40;
                List<String> list = new ArrayList<>();
                for (int i = s; i < 40 * page; i++) {
                    list.add("数据：" + i);
                }
                listData.postValue(list);
            }
        });
        s.start();
        return listData;
    }
}

File: kotlinTest/src/main/java/com/example/aac/kotlin/viewmodel/TestDataViewModel.kt
package com.example.aac.kotlin.viewmodel

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.os.SystemClock
import android.util.Log
/**
 * Created by yangc on 2017/8/14.
 * E-Mail:yangchaojiang@outlook.com
 * Deprecated:
 */

class TestDataViewModel(application: Application) : AndroidViewModel(application) {
    private val data = MutableLiveData<String>()
    private val listData = MutableLiveData<List<String>>()
    init {
        Log.d("TestDataViewModel", "TestDataViewModel")
    }
    fun getData(): LiveData<String> {
        Log.d("TestDataViewModel", data.value + "")
        val s = Thread(Runnable {
            SystemClock.sleep(1000)
            data.postValue("撒的阿萨德")
        })
        s.start()
        return data
    }

    override fun onCleared() {
        super.onCleared()
        Log.d("TestDataViewModel", "onCleared")
    }

    fun getListData(page: Int): LiveData<List<String>> {
        val s = Thread(Runnable {
            SystemClock.sleep(2000)
            val s = (page - 1) * 40
            val list = (s..40 * page - 1).map { "数据：" + it }
            listData.postValue(list)
        })
        s.start()
        return listData
    }
}
